<?xml version="1.0" encoding="utf-8"?>
<launch>

  <include file="$(find recog_pkg)/launch/png.launch" />
  <arg name="camera" default="head_rgbd_sensor_remote"/>
  <arg name="manager" default="/hsrb/head_rgbd_sensor/head_rgbd_sensor_nodelet_manager" />

  <include file="$(find openni2_launch)/launch/openni2.launch">
    <arg name="camera" value="$(arg camera)" />
    <arg name="rgb_frame_id" value="head_rgbd_sensor_rgb_frame" />
    <arg name="depth_frame_id" value="head_rgbd_sensor_rgb_frame" />
    <arg name="publish_tf" value="false" />
    <arg name="load_driver" value="false" />
    <arg name="depth_registration" value="true" />
    <arg name="queue_size" value="1000" />
  </include>

  <node name="relay_rgb_info"
	pkg="topic_tools" type="relay"
	args="/hsrb/head_rgbd_sensor/rgb/camera_info
	      head_rgbd_sensor_remote/rgb/camera_info">
  </node>

  <node name="relay_depth_info" 
	pkg="topic_tools" type="relay"
	args="/hsrb/head_rgbd_sensor/depth_registered/camera_info
	      head_rgbd_sensor_remote/depth_registered/camera_info">
  </node>

  <!-- <node name="relay_depth_image" pkg="nodelet" type="nodelet" -->
  <!-- 	args="load jsk_topic_tools/Relay $(arg manager)"> -->
  <!--   <remap from="~input" to="$(arg camera)/depth_registered/hw_registered/image_rect" /> -->
  <!--   <remap from="~output" to="$(arg camera)/depth_registered/image_rect" /> -->
  <!-- </node> -->

  <node name="rgb_decompress" pkg="image_transport" type="republish"
	args="compressed raw">
    <remap from="in" to="/hsrb/head_rgbd_sensor/rgb/image_raw" />
    <remap from="out" to="head_rgbd_sensor_remote/rgb/image_raw" />
  </node>

  <node name="depth_decompress" pkg="image_transport" type="republish"
	args="compressedDepth raw">
    <remap from="in" to="/hsrb/head_rgbd_sensor/depth_registered/image_raw"/>
    <remap from="out" to="head_rgbd_sensor_remote/depth_registered/image_raw" />
  </node>
</launch>
